Soft link SafariSafeBrowsing to resolve internal build dependency

pascoej opened the pull request:
#### c62dfd528744ec327cdc5db5d8569f42a88ecf0a
<pre>
Soft link SafariSafeBrowsing to resolve internal build dependency
<a href="https://rdar.apple.com/163584403">rdar://163584403</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=301586">https://bugs.webkit.org/show_bug.cgi?id=301586</a>

Reviewed by Elliott Williams.

Temporarily soft link SafariSafeBrowsing instead of weak linking to break
a build cycle.

* Source/WebKit/Configurations/WebKit.xcconfig:
Remove WK_SAFE_BROWSING_LDFLAGS definitions and usage.
* Source/WebKit/PlatformMac.cmake:
Remove -weak_framework SafariSafeBrowsing.
* Source/WebKit/SourcesCocoa.txt:
Add SafeBrowsingSoftLink.mm to unified sources.
* Source/WebKit/UIProcess/Cocoa/BrowsingWarningCocoa.mm:
Import SafeBrowsingSoftLink.h and remove hardcoded constants.
* Source/WebKit/UIProcess/Cocoa/SafeBrowsingSoftLink.h: Added.
* Source/WebKit/UIProcess/Cocoa/SafeBrowsingSoftLink.mm: Added.
Soft link macros for SafariSafeBrowsing framework.
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::beginSafeBrowsingCheck):
Use soft linked SSBLookupContext class.

Canonical link: <a href="https://commits.webkit.org/303307@main">https://commits.webkit.org/303307@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/55dbef557fd538b7419c29aa40c6479c0ff0bc5e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131901 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/4393 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/42911 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139413 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/83807 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/94e797f6-0ea5-4db4-802d-d2e1ffe805e5) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/133771 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/4333 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/4155 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/100823 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/68235 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/e6499f2d-1b9a-49e7-b771-7482ad0c72ff) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134847 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/3096 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/118135 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81614 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/e0be8963-426b-4fa6-ab37-eef18e6c9fb3) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/2986 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/843 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82633 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/111734 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/36259 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/142057 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/4063 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/36838 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/109197 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/4144 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/3558 "Found 1 new test failure: imported/w3c/web-platform-tests/webvtt/rendering/cues-with-video/processing-model/evil/media_height_19.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109363 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27715 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/3086 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114415 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/57284 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/4116 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/32815 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/3948 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/67563 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/4208 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/4076 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/SafeBrowsing.mm:
These changes don't seem strictly necessary, but seems nice to soft-link the same way.

pascoej replied:
If we don't softlink the same way in tests, the swizzling won't work.

emw-apple commented on Source/WebKit/UIProcess/Cocoa/BrowsingWarningCocoa.mm:
```diff
@@ -27,16 +27,13 @@
 #import <wtf/StdLibExtras.h>
 #import <wtf/text/MakeString.h>
 
-NSString * const SSBProviderGoogle = @"SSBProviderGoogle";
-NSString * const SSBProviderTencent = @"SSBProviderTencent";
-NSString * const SSBProviderApple = @"SSBProviderApple";
-
```
So it used to be that these strings were (re)declared in WebKit, and were also exported as constants from SSB? I would have expected that to cause a duplicate symbol error, but maybe we got away with it because these declarations are in an objc++ file so their names got mangled.

emw-apple commented on Source/WebKit/UIProcess/Cocoa/SafeBrowsingSoftLink.mm:
```diff
@@ -23,30 +23,20 @@
-public:
-    static Ref<RemoteMonotonicTimeline> create(TimelineID, const Seconds& originTime, MonotonicTime now);
+SOFT_LINK_CLASS_FOR_SOURCE(WebKit, SafariSafeBrowsing, SSBLookupContext);
+SOFT_LINK_CLASS_FOR_SOURCE(WebKit, SafariSafeBrowsing, SSBLookupResult);
+SOFT_LINK_CLASS_FOR_SOURCE(WebKit, SafariSafeBrowsing, SSBServiceLookupResult);
```
Do we need to update usage of these with `get<>ClassSingleton()`?

pascoej replied:
No usages other than the types.

pascoej committed:
Soft link SafariSafeBrowsing to resolve internal build dependency
rdar://163584403
https://bugs.webkit.org/show_bug.cgi?id=301586

Reviewed by Elliott Williams.

Temporarily soft link SafariSafeBrowsing instead of weak linking to break
a build cycle.

* Source/WebKit/Configurations/WebKit.xcconfig:
Remove WK_SAFE_BROWSING_LDFLAGS definitions and usage.
* Source/WebKit/PlatformMac.cmake:
Remove -weak_framework SafariSafeBrowsing.
* Source/WebKit/SourcesCocoa.txt:
Add SafeBrowsingSoftLink.mm to unified sources.
* Source/WebKit/UIProcess/Cocoa/BrowsingWarningCocoa.mm:
Import SafeBrowsingSoftLink.h and remove hardcoded constants.
* Source/WebKit/UIProcess/Cocoa/SafeBrowsingSoftLink.h: Added.
* Source/WebKit/UIProcess/Cocoa/SafeBrowsingSoftLink.mm: Added.
Soft link macros for SafariSafeBrowsing framework.
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::beginSafeBrowsingCheck):
Use soft linked SSBLookupContext class.

Canonical link: https://commits.webkit.org/303307@main
